main.go

```diff
@@ -258,6 +258,42 @@ func ShowVersion() {
 	os.Stdout.Write([]byte(v))
 }
 
+func doDaemon(config *menderConfig, dev *device, env *uBootEnv, opts *runOptionsType) error {
```

Comment:
I would call this one `initializeDaemon()` and `return NewDaemon(controller)`. Then in `main()` we would have:

```
daemon, err := initializeDaemon()
if err != nil {
  return daemon.Run()
}
```

Seems more descriptive than bit cryptic `doDaemon()`. What do you think?